skypy/galaxy/luminosity.py

```diff
@@ -110,23 +110,10 @@ def herbel_luminosities(redshift, alpha, a_m, b_m, size=None,
         size = np.shape(redshift)
 
     luminosity_star = _calculate_luminosity_star(redshift, a_m, b_m)
-    q = np.logspace(np.log10(np.min(q_min)), np.log10(np.max(q_max)),
-                    resolution)
-    cdf = _cdf(q, np.min(q_min), np.max(q_max), alpha)
-    t_lower = np.interp(q_min, q, cdf)
-    t_upper = np.interp(q_max, q, cdf)
-    u = np.random.uniform(t_lower, t_upper, size=size)
-    q_sample = np.interp(u, cdf, q)
-    luminosity_sample = luminosity_star * q_sample
-    return luminosity_sample
-
-
-def _cdf(q, q_min, q_max, alpha):
-    a = special.upper_incomplete_gamma(alpha+1, q_min)
-    b = special.upper_incomplete_gamma(alpha+1, q)
-    c = special.upper_incomplete_gamma(alpha+1, q_min)
-    d = special.upper_incomplete_gamma(alpha+1, q_max)
-    return (a-b)/(c-d)
+
+    x_sample = schechter(alpha, x_min, x_max, resolution=100, size=None)
```

Comment:
This should fix the unit test failing:
```suggestion
    x_sample = schechter(alpha, x_min, x_max, resolution=100, size=size)
```